server.js

```diff
@@ -1,33 +1,147 @@
+    success: true
+  })
+})
+
+// route provides a sorted list of all countries that are in netflixData
+app.get('/countries', (req, res) => {
+  // a list of all countries in the Netflix-Data without any duplicates. Therefore I am creating a Set (object with unique items). I convert the set into an array afterwards and sort it alphabetically
+  const countries = Array.from(new Set(netflixData.map((item) => item.country))).sort()
+
+  res.json({
+    response: countries,
+    success: true
+  })
+})
+
+// route with all shows (both movies and other) from the provided country
+app.get('/countries/:country', (req, res) => {
+  // req is an object with properties such as query and params, 
+  // both query and params are empty objects from the beginning 
+  // every query parameter consists of key and value
+  const { country } = req.params
+  const contentByCountry = netflixData.filter((item) => item.country.toLowerCase() === country)
+
+  if (contentByCountry.length === 0) {
+    res.status(404).json({
+      response: 'Sorry, but there is no content for this country',
+      success: false
+    })
+  } else {
+    res.json({
+      response: contentByCountry,
+      success: true
+    })
+  } 
+})
+
+// route provides all movies and has the possibility to query for director, year and actor. You can also do pagination by setting page & limit as query parameters
+app.get('/movies', (req, res) => {
+  const { director, year, actor, page, limit } = req.query
+
+  let filteredMovies = movies
+
+  if (director) {
+    filteredMovies = filteredMovies.filter((item) => item.director.toLowerCase().includes(director.toLowerCase()))
+  }
+
+  if (year) {
+    filteredMovies = filteredMovies.filter((item) => item.release_year === +year)
+  }
+
+  if (actor) {
+    filteredMovies = filteredMovies.filter((item) => item.cast.toLowerCase().includes(actor.toLowerCase()))
+  }
+
+  // PAGINATION
+  // limit = no. of items to show per page 
+  if (page && limit) {
+    // first index-no of item from filteredMovies we want to display
+    const startIndex = (page - 1) * limit
+    // last index-no of item from filteredMovies we want to display
+    const endIndex = page * limit 
+    // slice gives us what is in between startIndex and endIndex
+    const paginationArray = filteredMovies.slice(startIndex, endIndex)
+   
+    res.json(paginationArray)
+  }  
+
+  res.json({
+    response: filteredMovies,
+    success: true
+  })
+})
+
+// route provides one movie by ID
+app.get('/movies/id/:id', (req, res) => {
+  const { id } = req.params
+
+  const movie = movies.find((item) => item.show_id === +id)
+
+  if (!movie) {
+    res.status(404).json({
+      response: 'No movie found, that matches this ID',
+      success: false
+    })
+  } else {
+    res.json({
+      response: movie,
+      success: true
+    })
+  }
+})
+
+// route provides movies by name (can return more than one movie, if the provided parts of the title match with several movies)
```

Comment:
really nice way to let the frontender use this array!!! Like if you want to find all film with a word in its title!!!! this is a setup up compare to what i done in my project!